typescript/api/controllers/DynamicAssetController.ts

```diff
@@ -145,11 +146,17 @@ export module Controllers {
     * @param req
     * @param res
     */
-    public getAdminReportTemplates(req, res) {
+    public async getAdminReportTemplates(req, res) {
+      const brand: BrandingModel = BrandingService.getBrand(req.session.branding);
       const reportName = req.param("reportName") || "";
-      // TODO:
-      const entries = [];
-      return this.sendClientMappingJavascript(res, entries);
+
+      try {
+        const entries = await ReportsService.extractReportTemplates(brand, reportName);
+        return this.sendClientMappingJavascript(res, entries);
+      } catch (error) {
+        sails.log.error("Could not build report templates:", error);
+        return res.serverError();
```

Comment:
Probably not for right now, but instead of `res.serverError()`, use the API-aware base controller method, e.g. `this.sendResp(req, res, {status: 500, errors: [error]});`

The `sendResp` method may need a new feature to support the error modes for responding with compiled js.